packages/runtime/src/widgets/Form/TextInput.tsx

```diff
@@ -58,6 +58,28 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
     [action],
   );
 
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (values?.inputType !== "number") {
+        return;
+      }
+
+      const isNumericKey = /^\d$/.test(e.key); // Numpad keys (96-105)
+      const isControlKey = [
+        "Backspace",
+        "Tab",
+        "ArrowLeft",
+        "ArrowRight",
+      ].includes(e.key);
+      const isCopyPasteCut = e.ctrlKey || e.metaKey;
```

Comment:
This may be a better solution than checking for keys individually https://stackoverflow.com/a/70401792